### PR TITLE
feat: drawer component

### PR DESCRIPTION
**Pull request by chrisolsen**
# Before (the change)

# After (the change)

## Make sure that you've checked the boxes below before you submit the PR

- [x] I have read and followed the [setup steps](https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3086385489/Contribution+-+web+content#%5BinlineExtension%5DCode-contributions)
- [x] I have created necessary unit tests
- [x] I have tested the functionality in both React and Angular.

### React
```
import { GoADrawer } from "@abgov/react-components";
import { useState } from "react";

export function Drawer() {
  const [open, setOpen] = useState<boolean>(false);

  function toggleOpen() {
    setOpen(!open)      ;
  }
  
  return (
    <>
      <button onClick={toggleOpen}>Show</button>
      <GoADrawer open={open} position="left" maxSize="400px" onClose={toggleOpen}>
        <div style={{margin: "50rem 0"}}>
          content
        </div>
      </GoADrawer>
    </>
  )
}

export default Drawer;

```
### Angular
```
  <goa-button (_click)="toggleDrawer()">Menu</goa-button>
  <goa-drawer [open]="showDrawer" (_close)="toggleDrawer()" position="left" maxsize="400px" heading="Menu">
    <goa-accordion heading="Heading" headingsize="medium">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
    </goa-accordion>

    <div style="background-color: #fafafa; padding: 80rem 3rem">
      here
    </div>

    <goa-button-group alignment="start" slot="actions-foo">
      <goa-button (_click)="toggleDrawer()">Save</goa-button>
    </goa-button-group>
  </goa-drawer>

```

Add the following into the Angular component ts file
```
  showDrawer = false;
  toggleDrawer() {
    this.showDrawer = !this.showDrawer;
  }
```

Default token values 
```
    --goa-drawer-overlay-color: rgba(0,0,0,0.3);
    --goa-drawer-bottom-shadow: 0 -4px 12px rgba(0,0,0,0.2);
    --goa-drawer-left-shadow: 4px 0 12px rgba(0,0,0,0.2);
    --goa-drawer-right-shadow: -4px 0 12px rgba(0,0,0,0.2);
    --goa-drawer-transition-time: 200ms;
    --goa-drawer-padding: --goa-space-xl;

```

### CODE REVIEW

**Comment by lizhuomeng71** [comment]
There is minor gap when there is scroll bar for left and right drawer

image

**Comment by lizhuomeng71** [comment]
React Drawer Compoment give me this error message

image

File: libs/web-components/src/components/drawer/Drawer.svelte [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      <section class="drawer-actions" data-testid="drawer-actions">
+        <slot name="actions" />
+      </section>
+    {/if}
+  </div>
+</div>
+
+<style>
+  :host * {
+    --goa-drawer-bg-color: rgba(0,0,0,0.3);
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/web-components/src/components/drawer/Drawer.svelte:141]: This PR will have to wait until @twjeffery has the new design tokens finished and then we can sync things up after that.

**Comment by lizhuomeng71** [comment]
this feature is verified on react and angular

**Review by syedszeeshan — APPROVED** [review]

**Pull request merged**